core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java

```diff
@@ -747,6 +759,16 @@ public long getPageLengthInBytes() throws IOException {
    */
   public byte[] compress(Compressor compressor) throws IOException {
     DataType dataType = columnPageEncoderMeta.getStoreDataType();
+
+    // if the columnpage is isUnsafeEnabled and the Datatype is primitive.
+    // we try to compress the data in offheap directly, avoiding a copy from offheap to heap
+    if (isUnsafeEnabled() && (dataType == DataTypes.BOOLEAN || dataType == BYTE
+        || dataType == SHORT || dataType == DataTypes.SHORT_INT || dataType == INT
+        || dataType == LONG || dataType == FLOAT || dataType == DOUBLE
+        || DataTypes.isDecimal(dataType))) {
```

Comment:
is Decimal supported ? 

below I see getByteBufferArrayPage is unsupported in DecimalColumnPage